### PR TITLE
ExtendedVariantWidget Code cleanup

### PR DESCRIPTION
**Pull request by bbimber**
This PR does a handful of things:

- Removes a lot of unused imports and variables
- var -> const/let
- If the VCF has no genotypes (like some from the Demo session), the allele frequency table and genotype graphs failed. This now skips those sections if there are no genoypes.
- Question: is there a reason to use import vs. require()? Before this PR both were being used.
- I had intellij reformat the code for consistent indentation

### CODE REVIEW

**Comment by bbimber** [comment]
Also, I get this react-devotols warning when I open the details view:

```
react_devtools_backend.js:2842 Warning: Each child in a list should have a unique "key" prop.

Check the top-level render call using <WithStyles(ForwardRef(TableBody))>. See https://fb.me/react-warning-keys for more information.
    in WithStyles(ForwardRef(TableRow))
    in NewTable (created by wrappedComponent)
    in Suspense (created by wrappedComponent)
    in wrappedComponent (created by wrappedComponent)
    in div (created by ForwardRef(Paper))
    in ForwardRef(Paper) (created by WithStyles(ForwardRef(Paper)))
    in WithStyles(ForwardRef(Paper)) (created by wrappedComponent)
    in div (created by ForwardRef(Paper))
    in ForwardRef(Paper) (created by WithStyles(ForwardRef(Paper)))
    in WithStyles(ForwardRef(Paper)) (created by ForwardRef(Dialog))
    in div (created by Transition)
    in Transition (created by ForwardRef(Fade))
    in ForwardRef(Fade) (created by Unstable_TrapFocus)
    in Unstable_TrapFocus (created by ForwardRef(Modal))
    in div (created by ForwardRef(Modal))
    in ForwardRef(Portal) (created by ForwardRef(Modal))
    in ForwardRef(Modal) (created by ForwardRef(Dialog))
    in ForwardRef(Dialog) (created by WithStyles(ForwardRef(Dialog)))
    in WithStyles(ForwardRef(Dialog)) (created by wrappedComponent)
    in wrappedComponent (created by wrappedComponent)
    in div (created by wrappedComponent)
    in wrappedComponent (created by View)
    in ThemeProvider (created by View)
    in View
overrideMethod @ react_devtools_backend.js:2842
printWarning @ react.development.js:316
error @ react.development.js:288
validateExplicitKey @ react.development.js:1631
validateChildKeys @ react.development.js:1657
createElementWithValidation @ react.development.js:1807
eval @ ExtendedVariantWidget.js:284
commitHookEffectListMount @ react-dom.development.js:19732
commitPassiveHookEffects @ react-dom.development.js:19770
callCallback @ react-dom.development.js:189
invokeGuardedCallbackDev @ react-dom.development.js:238
invokeGuardedCallback @ react-dom.development.js:293
flushPassiveEffectsImpl @ react-dom.development.js:22854
unstable_runWithPriority @ scheduler.development.js:654
runWithPriority$1 @ react-dom.development.js:11040
flushPassiveEffects @ react-dom.development.js:22821
performSyncWorkOnRoot @ react-dom.development.js:21738
eval @ react-dom.development.js:11090
unstable_runWithPriority @ scheduler.development.js:654
runWithPriority$1 @ react-dom.development.js:11040
flushSyncCallbackQueueImpl @ react-dom.development.js:11085
flushSyncCallbackQueue @ react-dom.development.js:11073
discreteUpdates$1 @ react-dom.development.js:21894
discreteUpdates @ react-dom.development.js:807
dispatchDiscreteEvent @ react-dom.development.js:4169
```

Can you look into this error and see if we should be assigning a unique key to something?

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js [hunk 1]
```diff
@@ -138,7 +126,10 @@ export default jbrowse => {
         return displayJSX
     }
 
-    function makeChart(samples, ref, alt, classes){
+    function makeChart(samples, feat, classes){
```
Comment (jctrotter) [inline jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js:129]: This is a good change, but is outside the scope of general code cleanup. In the future changes like this should be in their own commits for the purposes of tracking development progress and for reverting changes if needed later down the line. Ok for now though. #63

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js [hunk 2]
```diff
@@ -242,15 +233,16 @@ export default jbrowse => {
...
-            var end = 0
-            var link = ActionURL.buildURL("jbrowse", "genotypeTable.view", null, {trackId: trackId, chr: contig, start: start, stop: end})
-            var href = <a href={link}>Click here to view sample-level genotypes</a>
+            const trackId = 0
+            const contig = feat["CHROM"];
+            const start = feat["POS"];
+            const end = feat["end"];
+
+            const link = ActionURL.buildURL("jbrowse", "genotypeTable.view", null, {trackId: trackId, chr: contig, start: start, stop: end})
+            const href = <a href={link} target="_blank">Click here to view sample-level genotypes</a>
```
Comment (jctrotter) [inline jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js:250]: See above. #63

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/fields.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
 export const fields = {
+    AF: {
+        title: 'Allele Frequency'
+    },
+    AC: {
+        title: 'Allele Count'
```
Comment (jctrotter) [inline jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/fields.js:6]: See above.

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js [hunk 4]
```diff
@@ -118,26 +118,31 @@ export default jbrowse => {
...
+                        {propertyJSX[i]}
+                    </BaseCard>
             )
         }
         return displayJSX
     }
 
     function makeChart(samples, feat, classes){
+        // Abort if there are no samples
+        if (Object.keys(samples).length === 0) {
```
Comment (jctrotter) [inline jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js:131]: See above.

**Comment by jctrotter** [comment]
Regarding require vs import, I believe I initially copied the use of require from an example, but I cannot find which example I used so I'll chalk it up to a holdover from some earlier jbrowse learning. I believe for our purposes either is functionally the same, and import is probably better practice. Good change.

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js (hunk 4 above)
Comment (bbimber) [reply jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js:131]: @jctrotter I'm fixing a bug in your code. no question this falls within the scope of code cleanup.

**Comment by jctrotter** [comment]
I will look into the warning here before putting an issue out. It's an issue independent of this PR, so we should be good on that end here.

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js (hunk 1 above)
Comment (bbimber) [reply jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/ExtendedVariantWidget.js:129]: This is not exactly a huge change and it's a lot more important that the software work robustly. The PR can be re-titled 'code cleanup and bugfixing'

**Pull request merged**